### PR TITLE
Add test code for unchecked scope (#203)

### PR DESCRIPTION
**Pull request by wonsubkim**
- unchecked  scope keyword clears checked function specifier
  check unchecked compound statement with checked function specifier

- unchecked scope clears checked property & prevents inheritance of checked property
  check unchecked block with nested checked block (compound statement, structure)

### CODE REVIEW

File: tests/typechecking/checked_scope.c [hunk 1]
```diff
@@ -275,6 +275,183 @@ void func30(void) {
...
+  checked {
+    int c checked[5][5];
+    int d[5][5];    // expected-error {{variable cannot have an unchecked array type}}
+    for (int i = 0; i < 5; i++) {
+      for (int j = 0; j < 5; j++) {
+        b[i][j] += c[i][j] - d[i][j];   // expected-error {{cannot use a variable with an unchecked type}}
+      }
+    }
+  }
+  return &b[4][4];  // expected-warning {{address of stack memory}}
```
Comment (dtarditi) [inline tests/typechecking/checked_scope.c]: Please update this as well.

File: tests/typechecking/checked_scope.c [hunk 2]
```diff
@@ -275,6 +275,183 @@ void func30(void) {
...
+checked int * func40(void) unchecked {  // expected-error {{return cannot have an unchecked pointer type}}
+  int a = 5;
+  ptr<int> pa = &a;
+  int b[5][5];
+  for (int i = 0; i < 5; i++) {
+    for (int j = 0; j < 5; j++) {
+      b[i][j] = -1;
+    }
+  }
+  return &b[4][4];  // expected-warning {{address of stack memory}}
```
Comment (dtarditi) [inline tests/typechecking/checked_scope.c]: Could you change this to return 0 (NULL) instead?     It is not correct to return the address of a local variable.

File: tests/typechecking/checked_scope.c [hunk 3]
```diff
@@ -275,6 +275,183 @@ void func30(void) {
...
+      ptr<int> pa = &a;
+      int b checked[5][5];
+      for (int i = 0; i < 5; i++) {
+        for (int j = 0; j < 5; j++) {
+          b[i][j] = -1;
+        }
+      }
+    }
+  }
+  return &a;  // expected-warning {{address of stack memory}}
```
Comment (dtarditi) [inline tests/typechecking/checked_scope.c]: Please update this return 0 (NULL) instead.

File: tests/typechecking/checked_scope.c [hunk 4]
```diff
@@ -275,6 +275,183 @@ void func30(void) {
   checked [5][5];   // expected-error {{expected compound statement after checked scope keyword}}
 }
 
+// unchecked scope test
+// - unchecked scope clears checked property & prevents inheritance of checked property
```
Comment (dtarditi) [inline tests/typechecking/checked_scope.c]: Please add periods at the ends of these sentences.

**Comment by wonsubkim** [comment]
I missed unchecked function specifier.
Since I thought that function can't be defined in checked scope, unchecked function specifier is not necessary
unchecked function specifier is meaningful only if it is used in checke scope.

How about this issue?

**Comment by wonsubkim** [comment]
I am ready for review

**Comment by dtarditi** [comment]
I'm going to OK this and open a new issue to review the code for improving the testing for the BOUNDS_CHECKED pragma.

**Comment by wonsubkim** [comment]
Yes, Pragma can be turned on and off
For now, I have assumed that current BOUNDS_CHECKED pragma can be applied into only top level scope
I will consider further issues

**Pull request merged**